src/main/kotlin/org/rust/lang/core/psi/ext/RsTypeAlias.kt

```diff
@@ -20,6 +20,16 @@ import javax.swing.Icon
 val RsTypeAlias.default: PsiElement?
     get() = node.findChildByType(DEFAULT)?.psi
 
+fun RsTypeAlias.baseType(): RsElement? {
+    var base: RsElement? = this
+    val visited = mutableSetOf<RsElement>(this)
```

Comment:
This newly extracted method seems to prevent endless loops when a type aliases form a cycle.
Can you also update the very similar code in `TypeInference.kt` `inferStructLiteralType()` to do the same?
If this loop prevention is intentional, I guess it would also make sense to add tests for it and to mention it in the commit message?